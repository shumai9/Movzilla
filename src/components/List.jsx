import React, {Component} from 'react';
import '../style/List.css'

class List extends Component{  
  render(){
    const data = this.props.movies;
    return(
      <div className="movie-list">
        <h1>Top Featured Movies</h1>
          <ul>
          { data ?
            (Object.keys(data).map((line) =>{
              return(
                <li key={line}>
                  <a href="#">
                    {data[line].Title}
                  </a>
                </li>
              ) 
            })
            ) : (
              null
            )
          }
        </ul>
      </div>
    );
  }
}

export default List;
