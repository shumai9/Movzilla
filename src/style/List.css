.list{
  min-width: 25%;
}
.list li, a {
   color: #000;
   list-style: none;
   text-decoration: none;
   text-align: left; 
}