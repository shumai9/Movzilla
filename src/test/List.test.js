import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from '../components/List';

it('renders with out  failing', () =>{
  const div = document.createElement('div');
  ReactDOM.render(<List />, div);
  ReactDOM.unmountComponentAtNode(div);
});
it('renders correctly', () =>{
  const tree = renderer
    .create(<List />)
    .toJSON();
  expect(tree).toMatchSnapshot();
}); 
  