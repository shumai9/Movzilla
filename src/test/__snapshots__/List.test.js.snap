// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="movie-list"
>
  <h1>
    Top Featured Movies
  </h1>
  <ul />
</div>
`;
